pkg/webhookconfig/common.go

```diff
@@ -75,12 +76,17 @@ func (wrc *Register) constructOwner() v1.OwnerReference {
 }
 
 func (wrc *Register) GetKubePolicyClusterRoleName() (*unstructured.Unstructured, error) {
-	clusterRole, err := wrc.client.ListResource(config.ClusterRoleAPIVersion, config.ClusterRoleKind, "", &v1.LabelSelector{MatchLabels: map[string]string{"app.kubernetes.io/created-by": "garbageCollector"}})
+	clusterRoles, err := wrc.client.ListResource(config.ClusterRoleAPIVersion, config.ClusterRoleKind, "", &v1.LabelSelector{MatchLabels: map[string]string{"app": "kyverno"}})
 	if err != nil {
 		return nil, err
 	}
-
-	return &clusterRole.Items[0], nil
+	var clusterRole unstructured.Unstructured
+	for _, cr := range clusterRoles.Items {
+		if strings.HasSuffix(cr.GetName(), "webhook") {
+			clusterRole = cr
+		}
+	}
+	return &clusterRole, nil
```

Comment:
If there is no match, should we return an error?

How does the callee handle the case when the role is not found?

Reply:
Let me handle the scenario.